v0/destinations/kustomer/transform.js

```diff
@@ -0,0 +1,233 @@
+} = require("../../util");
+const { fetchKustomer, validateEvent } = require("./util");
+
+// Function responsible for constructing the Kustomer (User) Payload for identify
+// type of events.
+const constructKustomerPayload = (message, category) => {
+  const kustomerPayload = constructPayload(
+    message,
+    MAPPING_CONFIG[category.name]
+  );
+
+  if (
+    !get(kustomerPayload, "name") &&
+    getFieldValueFromMessage(message, "firstName") &&
+    getFieldValueFromMessage(message, "lastName")
+  ) {
+    kustomerPayload.name = `${getFieldValueFromMessage(
+      message,
+      "firstName"
+    )} ${getFieldValueFromMessage(message, "lastName")}`;
+  }
+
+  if (getFieldValueFromMessage(message, "email")) {
+    kustomerPayload.emails = kustomerPayload.emails
+      ? kustomerPayload.emails
+      : [{ type: "home", email: getFieldValueFromMessage(message, "email") }];
+  }
+
+  if (getFieldValueFromMessage(message, "phone")) {
+    kustomerPayload.phones = kustomerPayload.phones
+      ? kustomerPayload.phones
+      : [{ type: "home", phone: getFieldValueFromMessage(message, "phone") }];
+  }
+  if (getFieldValueFromMessage(message, "website")) {
+    kustomerPayload.urls = [
+      {
+        url: getFieldValueFromMessage(message, "website")
+      }
+    ];
+  }
+
+  const address = getFieldValueFromMessage(message, "address");
+  if (address) {
+    let addrStr;
+    if (typeof address === "string") {
+      addrStr = address;
+    } else {
+      const { street, city, state, postalCode } = address;
+      addrStr =
+        street && city && state && postalCode
+          ? `${street}, ${city}, ${state}, ${postalCode}`
+          : addrStr;
+    }
+    if (typeof addrStr === "string") {
+      kustomerPayload.locations = [
+        {
+          type: "home",
+          address: addrStr
+        }
+      ];
+    }
+  }
+
+  removeUndefinedAndNullValues(kustomerPayload);
+  return kustomerPayload;
+};
+
+// Main process function responsible for building payload for all
+// type of events.
+const responseBuilderSimple = async (message, category, destination) => {
+  let payload = {};
+  let targetUrl;
+  let userExists = false;
+  // In case of identify type of event first extract the anonymousId, userId
+  // and search if any Kustomer is present in destination.
+  // If present update the same kustomer with the given payload.
+  // else create a new kustomer.
+  //-------------------------------------------------------------
+  // Get Kustomer: https://apidocs.kustomer.com/#ff41f372-6144-4c64-9712-662ee5ef1c33
+  // Create Kustomer: https://apidocs.kustomer.com/#07bd1072-4d4b-4875-b526-8369d711e811
+  // Update Kustomer: https://apidocs.kustomer.com/#077d653a-184e-4153-8133-d24b6427c1ae
+  if (message.type.toLowerCase() == EventType.IDENTIFY) {
+    let response;
+    // If email exists we first search Kustomer with email if present then we mark it
+    // for update call.
+    if (getFieldValueFromMessage(message, "email")) {
```

Comment:
`getFieldValueFromMessage` method runs a loop. IMO, we can optimize by storing the value in a variable here.

Reply:
fixed